registries/adapters/partner_rhcc_adapter.go

```diff
@@ -0,0 +1,180 @@
+//
+// Copyright (c) 2018 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package adapters
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"github.com/automationbroker/bundle-lib/bundle"
+	log "github.com/sirupsen/logrus"
+)
+
+const partnerName = "partner_rhcc"
+const partnerManifestURL = "%v/v2/%v/manifests/%v"
+const partnerCatalogURL = "%v/v2/_catalog"
+
+// PartnerRhccAdapter - Partner RHCC Adapter
+type PartnerRhccAdapter struct {
+	Config Configuration
+}
+
+// PartnerCatalogResponse - Partner Catalog Response
+type PartnerCatalogResponse struct {
+	Repositories []string `json:"repositories"`
+}
+
+// RegistryName - Retrieve the registry name
+func (r PartnerRhccAdapter) RegistryName() string {
+	return partnerName
+}
+
+// GetImageNames - retrieve the images
+func (r PartnerRhccAdapter) GetImageNames() ([]string, error) {
+	log.Debug("PartnerRhccAdapter::GetImageNames")
+	log.Debugf("BundleSpecLabel: %s", BundleSpecLabel)
+
+	if r.Config.Images != nil {
+		log.Debugf("Configured to use images: %v", r.Config.Images)
+		return r.Config.Images, nil
+	}
+	log.Debugf("Did not find images in config, attempting to discover from %s/v2/_catalog", r.Config.URL)
+
+	req, err := http.NewRequest("GET", fmt.Sprintf(partnerCatalogURL, r.Config.URL), nil)
```

Comment:
I think this should use the change from @maleck13 PR #87 .

Reply:
It looks like @maleck13's change only affects the `loadSpec` function not the Catalog call... this is unique to the partner adapter. 

I'm fine with making the change to `loadSpec` once it's merged.